builtin_op_importers.cpp

```diff
@@ -1689,28 +1689,20 @@ DEFINE_BUILTIN_OP_IMPORTER(Slice) {
   const nvinfer1::Dims sliceStride = makeDims(1); // ONNX has no support for strides in Slice
   for (size_t i = 0; i < axes.size(); i++){
     int axis = axes[i];
-    // Special pass through for no-ops (slice across the whole dimension, [:])
-    if (starts[i] == 0 && ends[i] >= dims.d[i])
-    {
+    if (axis == 0) {
+      // We can only check that starts is properly 0
+      // but can't check end as we don't know batch size
+      ASSERT(starts[i] == 0; ErrorCode::kINVALID_VALUE);
```

Comment:
This is somewhat dangerous as we will treat (for example) a  [0:4,...] slice on a size [8,...] tensor as a no-op, which can give incorrect inference results. We should at least print a warning to the user that TensorRT sees a slice across the batch dimension with starts = 0 and is ignoring it.

Reply:
From what I understand, if the user want to slice on the batch axis, there is no way it will be working anyway as TRT doesn't even consider the BatchAxis (before my modification it would also silently crash I believe)

But yes, we can add a warning whenever we convert a slice. (start must be 0, so there will be either a warning either throw an error at the assertion)